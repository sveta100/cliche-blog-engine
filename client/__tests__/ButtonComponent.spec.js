import { mount } from '@vue/test-utils';
import Button from '../src/components/common/ButtonComponent.vue';

describe('Button component', () => {
  const onClick = jest.fn();
  const wrapper = mount(Button, {
    listeners: {
      click: onClick,
    },
  });

  test('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correct markup', () => {
    const btn = wrapper.find('button');
    expect(btn.exists()).toBe(true);
  });

  test('once clicked an event is emitted', () => {
    wrapper.trigger('click');
    expect(onClick).toHaveBeenCalled();
  });
});
