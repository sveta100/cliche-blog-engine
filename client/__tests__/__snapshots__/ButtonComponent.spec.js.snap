// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
"<button type=\\"button\\" class=\\"super-btn super-btn--default\\">
  Button
</button>"
`;
